app/javascript/components/shared/TopHeader.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import PropTypes from "prop-types";
+import TopHeaderDropdownMenu from "./TopHeaderDropdownMenu";
+
+class TopHeader extends React.Component {
+  render() {
+    const { username, isAdmin, profileLink, editUserLink, adminLink, logoutLink } = this.props;
+    const topHeaderDropdownMenuComponent = function() {
+      if (username) {
```

Comment:
It is not blocking, but a better way to remove else-null block is return (username && TopHeaderDropdownMenu)